src/Microsoft.ML.Transforms/FourierDistributionSampler.cs

```diff
@@ -5,205 +5,270 @@
+        /// its inputs. The <see cref="GaussianKernel"/> depends on the L2 distance, and the <see cref="LaplacianKernel"/> depends
+        /// on the L1 distance.
+        /// </summary>
+        internal abstract float Distance(in VBuffer<float> first, in VBuffer<float> second);
+
+        /// <summary>
+        /// This method returns an object that can sample from the non-negative measure that is the Fourier transform of this kernel.
+        /// </summary>
+        internal abstract FourierRandomNumberGeneratorBase GetRandomNumberGenerator(float averageDistance);
     }
 
-    [TlcModule.ComponentKind("FourierDistributionSampler")]
-    internal interface IFourierDistributionSamplerFactory : IComponentFactory<float, IFourierDistributionSampler>
+    /// <summary>
+    /// The Fourier transform of a continuous positive definite kernel is a non-negative measure
+    /// (<a href="https://en.wikipedia.org/wiki/Bochner%27s_theorem">Bochner's theorem</a>). This class
+    /// samples numbers from the non-negative measure corresponding to the given kernel.
+    /// </summary>
+    internal abstract class FourierRandomNumberGeneratorBase
     {
+        public abstract float Next(Random rand);
     }
 
-    public sealed class GaussianFourierSampler : IFourierDistributionSampler
+    public sealed class GaussianKernel : KernelBase
     {
-        private readonly IHost _host;
-
-        public sealed class Options : IFourierDistributionSamplerFactory
+        internal sealed class Options : IComponentFactory<KernelBase>
         {
             [Argument(ArgumentType.AtMostOnce, HelpText = "gamma in the kernel definition: exp(-gamma*||x-y||^2 / r^2). r is an estimate of the average intra-example distance", ShortName = "g")]
             public float Gamma = 1;
 
-            IFourierDistributionSampler IComponentFactory<float, IFourierDistributionSampler>.CreateComponent(IHostEnvironment env, float avgDist)
-                => new GaussianFourierSampler(env, this, avgDist);
-        }
-
-        internal const string LoaderSignature = "RandGaussFourierExec";
-        private static VersionInfo GetVersionInfo()
-        {
-            return new VersionInfo(
-                modelSignature: "RND GAUS",
-                verWrittenCur: 0x00010001, // Initial
-                verReadableCur: 0x00010001,
-                verWeCanReadBack: 0x00010001,
-                loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(GaussianFourierSampler).Assembly.FullName);
+            public KernelBase CreateComponent(IHostEnvironment env) => new GaussianKernel(env, this);
         }
 
         internal const string LoadName = "GaussianRandom";
 
         private readonly float _gamma;
 
-        public GaussianFourierSampler(IHostEnvironment env, Options options, float avgDist)
+        public GaussianKernel(float gamma = 1)
```

Comment:
>GaussianKernel [](start = 15, length = 14)

would be nice to have comment.
Also `MathUtils.Sqrt(2 * _gamma)`
Since I doubt average distance can be negative and _gamma in random generator is this gamma / avgDistance, we need to specify what gamma is non-negative number. and assert it.